lib/source.js

```diff
@@ -3,7 +3,8 @@ const { GREATEST_LOWER_BOUND, LEAST_UPPER_BOUND } = require('source-map').Source
 
 module.exports = class CovSource {
   constructor (sourceRaw, wrapperLength) {
-    sourceRaw = sourceRaw.trimEnd()
+    // Treat null sourceRaw as the empty string
+    sourceRaw = sourceRaw ? sourceRaw.trimEnd() : ''
```

Comment:
I think there could be a bad case: if a source file fails to read, but `''` is default value here, the data passed back to c8 may have an extra piece of useless data.

Reply:
For example, if you read a file `dist. js` that includes `dist.js.map` but  has no source file, the data passed to C8 might be `branch`, `function` and `line` with `0/0`.

Reply:
@bvanjoi This sounds like a feature to me.

Ignoring the files that were failed to read completely hides some potential problems.
We must have the user noticed in some ways.

If the failure is because it shouldn't be read, then the user should use `excludePath` function to exclude it instead of expect it to be automatically filtered out.

Reply:
@bvanjoi Filtering out the files that are not excluded by `excludePath` sounds more like a bug.